src/utils/indexer/getBlockTime.ts

```diff
@@ -0,0 +1,44 @@
+import { getSdk } from "../cache/getSdk";
+import { logger } from "../logger";
+
+const KNOWN_BLOCKTIME_SECONDS = {
+  1: 12,
+  137: 2,
+} as Record<number, number>;
+
+const DEFAULT_BLOCKTIME_SECONDS = 10;
+const BLOCKS_TO_ESTIMATE = 10;
+
+export const getBlockTimeSeconds = async (chainId: number) => {
+  if (KNOWN_BLOCKTIME_SECONDS[chainId]) {
+    return KNOWN_BLOCKTIME_SECONDS[chainId];
+  }
+
+  const sdk = await getSdk({ chainId });
+  const provider = sdk.getProvider();
+  try {
+    const latestBlockNumber = await provider.getBlockNumber();
+    const blocks = [];
+
+    for (let i = 0; i < BLOCKS_TO_ESTIMATE; i++) {
+      const block = await provider.getBlock(latestBlockNumber - 1);
+      blocks.push(block);
+    }
+
+    let totalTimeDiff = 0;
+    for (let i = 0; i < blocks.length - 1; i++) {
+      totalTimeDiff += blocks[i].timestamp - blocks[i + 1].timestamp;
+    }
+    const averageBlockTime = totalTimeDiff / (blocks.length - 1);
```

Comment:
@IDubuque 

this needs to be `i`. to get everything working.

Reply:
also, I m not able to get any Events for my [contract](https://mumbai.polygonscan.com/address/0x365b83d67d5539c6583b9c0266a548926bf216f4) on `mumbai`.

you you try and change the DB value for `lastIndexedBlock` to `46436824` and you should be seeings logs/events coming in and getting added to DB. but I don't see anything there

Reply:
Nice catch, missed this typo and only tested with hardcoded chains